src/lsp.rs

```diff
@@ -52,30 +59,54 @@ struct Backend<TLocalLinter> {
     client: Client,
     local_linter: TLocalLinter,
     per_file: Mutex<HashMap<Url, PerFileState>>,
+    start_new_trace_sender: Option<Sender<PathBuf>>,
+    per_config_context: RwLock<Option<PerConfigContext>>,
```

Comment:
I was thinking this could be a `OnceLock` but I don't know if its API supports "check if it's been populated"?